qa/workunits/rbd/rbd_groups.sh

```diff
@@ -210,15 +210,26 @@ check_snapshot_info()
     local snap_name=$2
     local image_count=$3
 
-    local snap_info=$(rbd group snap info $group_name@$snap_name --format=json)
-    local actual_snap_name=$(jq -r ".name" <<< "$snap_info")
+    local snap_info_json=$(
+        rbd group snap info $group_name@$snap_name --format=json)
```

Comment:
> rbd group snapshot 'snap4':
> 	id: 104cf0614061
> 	state: complete
> 	image snap: .group.1_103647dab2e1_104cf0614061
> 	images:
> 		rbd/img3 (snap id: 3)
@ajarr what are the other possible states for the group snapshots?
can we make it as `image snap` as `image snap name` 
also can we provide `Snap Size` and `Namespace` info here?

like below output 

> [root@ceph-sangadi-bz-c5lfcn-node4 ~]# rbd snap ls --all pool1/p1_image1
> SNAPID  NAME                                SIZE    PROTECTED  TIMESTAMP                 NAMESPACE
>      3  snap1                               10 GiB  yes        Mon Aug  5 05:11:38 2024  user
>      4  .group.7_68474491ae8b_6880eaae0d10  10 GiB             Mon Aug  5 05:12:38 2024  group (pool1/group1@p1g1snap1)
>      5  .group.7_68474491ae8b_699a6d77341d  10 GiB             Tue Aug  6 03:08:02 2024  group (pool1/group1@p1g1snap2)
>      7  .group.7_68474491ae8b_43951876cf01  10 GiB             Tue Aug  6 04:53:56 2024  group (pool1/group1@p1g1snap3)
>      9  .group.7_68474491ae8b_4509517fdd67  10 GiB             Wed Aug  7 05:22:21 2024  group (pool1/group1@p1g1snap4)
> [root@ceph-sangadi-bz-c5lfcn-node4 ~]#
> 

can you please add output of multiple group snap info?

Reply:
> > rbd group snapshot 'snap4':
> > id: 104cf0614061
> > state: complete
> > image snap: .group.1_103647dab2e1_104cf0614061
> > images:
> > rbd/img3 (snap id: 3)
> 
> @ajarr what are the other possible states for the group snapshots? 

The other state is `incomplete` . You can refer to,
https://github.com/ceph/ceph/blob/5622aec40af40714e40b45ad27fc0541b3c010c1/src/tools/rbd/action/Group.cc#L90

> can we make it as `image snap` as `image snap name` 

Hmm... I thought we all agreed upon the fields in the CLI output.  See our previous discussion here, 
https://github.com/ceph/ceph/pull/57797#issuecomment-2145004799
I like to avoid changing the field names in CLI outputs that we already discussed.

> also can we provide `Snap Size` and `Namespace` info here? > like below output
> 
> > [root@ceph-sangadi-bz-c5lfcn-node4 ~]# rbd snap ls --all pool1/p1_image1
> > SNAPID  NAME                                SIZE    PROTECTED  TIMESTAMP                 NAMESPACE
> > 3  snap1                               10 GiB  yes        Mon Aug  5 05:11:38 2024  user
> > 4  .group.7_68474491ae8b_6880eaae0d10  10 GiB             Mon Aug  5 05:12:38 2024  group (pool1/group1@p1g1snap1)
> > 5  .group.7_68474491ae8b_699a6d77341d  10 GiB             Tue Aug  6 03:08:02 2024  group (pool1/group1@p1g1snap2)
> > 7  .group.7_68474491ae8b_43951876cf01  10 GiB             Tue Aug  6 04:53:56 2024  group (pool1/group1@p1g1snap3)
> > 9  .group.7_68474491ae8b_4509517fdd67  10 GiB             Wed Aug  7 05:22:21 2024  group (pool1/group1@p1g1snap4)
> > [root@ceph-sangadi-bz-c5lfcn-node4 ~]#
> 

Similar question here https://github.com/ceph/ceph/pull/57797#issuecomment-2145262941 ? See Ilya's reply here, https://github.com/ceph/ceph/pull/57797#issuecomment-2145277949

You can fetch the namespace from either the image spec format in the human readable output, or better from the 'namespace' field in the JSON formatted output.

I'm not sure about 'snap size' of individual member images. The users are expected to use the `snap ls --all` like you've used to figure it out.

> can you please add output of multiple group snap info?
Not sure what you mean here?  Output of `rbd group snap info` for a gp snapshot with multiple member images. See,
https://github.com/ceph/ceph/pull/57797#issuecomment-2233172495

Reply:
ok thanks @ajarr for addressing it.